packages/commonwealth/client/scripts/views/modals/AuthModal/useAuthentication.tsx

```diff
@@ -322,24 +322,23 @@ const useAuthentication = (props: UseAuthenticationProps) => {
       // Important: when we first create an account and verify it, the user id
       // is initially null from api (reloading the page will update it), to correct
       // it we need to get the id from api
-      const updatedProfiles =
-        await DISCOURAGED_NONREACTIVE_fetchProfilesByAddress(
-          // @ts-expect-error <StrictNullChecks>
-          account.profile.chain,
-          // @ts-expect-error <StrictNullChecks>
-          account.profile.address,
-        );
-      const currentUserUpdatedProfile = updatedProfiles[0];
-      if (!currentUserUpdatedProfile) {
+      const userAddresses = await trpcVanillaClient.user.getUserAddresses.query(
```

Comment:
can't you use trpc here instead of trpcVanillaClient ?

Reply:
I just replaced an implementation that was marked as TO_BE_DEPRECATED with the vanilla option b/c it was in a place where hooks are not possible according to the comments. If you think we can still use hooks here, we can change it to the react query client

Reply:
in this particular case we are in the hook so we can also use hook. Thus, I would prefer to use existing `useFetchProfilesByAddressesQuery`.

Reply:
Historical context: There are a of branching cases for auth here, and local react state updates caused issues with fetching from react query or using local state as a reference (notice the functions are using passed arguments instead of react-state for correct "state" of a var).

I suggest we use the vanilla client as using react-query/trpc would require us to update local state (with the id of profile) we want to fetch and then again listen/wait for this response, then use a `useEffect` for further logic.

In short: lets use the current implementation.